src/models/tabularmodel.jl

```diff
@@ -0,0 +1,56 @@
+function emb_sz_rule(n_cat)
+	min(600, round(1.6 * n_cat^0.56))
+end
+
+function _one_emb_sz(catdict, catcol::Symbol, sz_dict=nothing)
+	sz_dict = isnothing(sz_dict) ? Dict() : sz_dict
+	n_cat = length(catdict[catcol])
+	sz = catcol in keys(sz_dict) ? sz_dict[catcol] : emb_sz_rule(n_cat)
+	Int64(n_cat), Int64(sz)
+end
+
+function get_emb_sz(catdict, cols; sz_dict=nothing)
+	[_one_emb_sz(catdict, catcol, sz_dict) for catcol in cols]
+end
+
+# function get_emb_sz(td::TableDataset, sz_dict=nothing)
+# 	cols = Tables.columnaccess(td.table) ? Tables.columnnames(td.table) : Tables.columnnames(Tables.rows(td.table)[1])
+# 	[_one_emb_sz(catdict, catcol, sz_dict) for catcol in cols]
+# end
+
+function TabularModel(
+        layers; 
+        emb_szs,
+        n_cont::Int64,
+        out_sz,
+        ps::Union{Tuple, Vector, Number}=0,
+        embed_p::Float64=0.,
+        y_range=nothing,
+        use_bn::Bool=true,
+        bn_final::Bool=false,
+        bn_cont::Bool=true,
+        act_cls=Flux.relu,
+        lin_first::Bool=true)
+
+    embedslist = [Embedding(ni, nf) for (ni, nf) in emb_szs]
+    emb_drop = Dropout(embed_p)
+    embeds = Chain(x -> ntuple(i -> x[i, :], length(emb_szs)), Parallel(vcat, embedslist...), emb_drop)
+
+    bn_cont = bn_cont ? BatchNorm(n_cont) : identity
+
+    n_emb = sum(size(embedlayer.weight)[1] for embedlayer in embedslist)
```

Comment:
This would be clearer with `Flux.outputsize`

Reply:
@manikyabard can you paste the error you got with `Embedding` and `outputsize`?

Reply:
Sure, so if I do something like this
```julia
Flux.outputsize(Embedding(2, 3), (2,1))
```
the error I get is
```
ArgumentError: unable to check bounds for indices of type Flux.NilNumber.Nil

Stacktrace:
  [1] checkindex(#unused#::Type{Bool}, inds::Base.OneTo{Int64}, i::Flux.NilNumber.Nil)
    @ Base ./abstractarray.jl:671
  [2] checkindex
    @ ./abstractarray.jl:686 [inlined]
  [3] checkbounds_indices (repeats 2 times)
    @ ./abstractarray.jl:642 [inlined]
  [4] checkbounds
    @ ./abstractarray.jl:595 [inlined]
  [5] checkbounds
    @ ./abstractarray.jl:616 [inlined]
  [6] _getindex
    @ ./multidimensional.jl:831 [inlined]
  [7] getindex
    @ ./abstractarray.jl:1170 [inlined]
  [8] Embedding
    @ ~/.julia/packages/Flux/wii6E/src/layers/basic.jl:421 [inlined]
  [9] (::Embedding{Matrix{Float32}})(x::Matrix{Flux.NilNumber.Nil})
    @ Flux ~/.julia/packages/Flux/wii6E/src/layers/basic.jl:422
 [10] #outputsize#279
    @ ~/.julia/packages/Flux/wii6E/src/outputsize.jl:93 [inlined]
 [11] outputsize(m::Embedding{Matrix{Float32}}, inputsizes::Tuple{Int64, Int64})
    @ Flux ~/.julia/packages/Flux/wii6E/src/outputsize.jl:92
 [12] top-level scope
    @ In[213]:1
 [13] eval
    @ ./boot.jl:360 [inlined]
 [14] include_string(mapexpr::typeof(REPL.softscope), mod::Module, code::String, filename::String)
    @ Base ./loading.jl:1094
```